dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/security/oidc/provider/GoogleProvider.java

```diff
@@ -52,12 +57,48 @@ private GoogleProvider()
         throw new IllegalStateException( "Utility class" );
     }
 
-    public static DhisOidcClientRegistration build( DhisConfigurationProvider config )
+    public static DhisOidcClientRegistration parse( Properties config )
     {
         Objects.requireNonNull( config, "DhisConfigurationProvider is missing!" );
 
-        String clientId = config.getProperty( OIDC_PROVIDER_GOOGLE_CLIENT_ID );
-        String clientSecret = config.getProperty( OIDC_PROVIDER_GOOGLE_CLIENT_SECRET );
+        Map<String, Set<String>> allKeysByProvider = GenericOidcProviderConfigParser
+            .extractKeysGroupByProvider( config );
+
+        Set<String> googleKeys = Optional.ofNullable( allKeysByProvider.get( REGISTRATION_ID ) )
+            .orElse( Collections.emptySet() );
+
+        if ( googleKeys.isEmpty() )
+        {
+            return null;
+        }
```

Comment:
Maybe I am just not an `Optional` fan, but I would find something boring like the below more communicating intent:
```java
if ( !allKeysByProvider.containsKey( REGISTRATION_ID ) || allKeysByProvider.get( REGISTRATION_ID ).isEmpty()) {
  return null;
}
Set<String> googleKeys = allKeysByProvider.get( REGISTRATION_ID );
```